classes/class-nets-easy-checkout.php

```diff
@@ -43,7 +43,7 @@ public function update_nets_easy_order( $cart ) {
 
 		$payment_id = WC()->session->get( 'dibs_payment_id' );
 		if ( empty( $payment_id ) ) {
-			return;
+			wp_send_json_error( 'Failed to update the Nexi order. Payment ID missing.' );
```

Comment:
This is does not only happen during an ajax request. So in this case returning a json error could potentially cause issues since that will print some json and then die. `woocommerce_after_calculate_totals` is triggered on a page load in some cases and with how the flow is right now, if for some reason we simply dont have a `payment_id` it would cause the page to not function. For example this is triggered when you add a product to the cart as well.

And since this runs on a hook, we have no way to tell what triggers it or why. There are also plugins that can be reliant on this hook to function properly, and this would potentially cause issues for them.

So this should be a simple return like the rest of the checks have here. The rule is to only use wp_send_json_error, wp_send_json_success or wp_send_json during an actual AJAX callback function. Any functions that are called in that processes should rather return something the ajax function can use to determine if they should return an error or success.